### PR TITLE
Mainnet release 1012 diff

### PR DESCRIPTION
**Pull request by rlan35**

### CODE REVIEW

File: core/types/block.go [hunk 1]
```diff
@@ -181,12 +181,13 @@ func init() {
...
+// Block represents an entire block in the Harmony blockchain.
 type Block struct {
-	header           *block.Header
-	uncles           []*block.Header
-	transactions     Transactions
-	incomingReceipts CXReceiptsProofs
+	header              *block.Header
+	uncles              []*block.Header
+	transactions        Transactions
+	stakingTransactions staking.StakingTransactions
```
Comment (astralblue) [inline core/types/block.go]: TODO @fxfactorial – add corresponding fields to extblock; use the versioning scheme (extblockV3)

File: msgq/msgq.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// Close closes the given queue.
+func (q *Queue) Close() error {
+	close(q.ch)
+	return nil
+}
+
+// ErrRxOverrun signals that a receive queue has been overrun.
+var ErrRxOverrun = errors.New("rx overrun")
```
Comment (astralblue) [inline msgq/msgq.go]: TODO @harmony-ek – collect metrics (queue length, overrun counter)

File: node/node.go [hunk 3]
```diff
@@ -399,6 +409,21 @@ func (node *Node) getTransactionsForNewBlock(
...
+	for i := 0; i < RxWorkers; i++ {
+		go node.rxQueue.HandleMessages(node)
+	}
+
+	// start the goroutine to receive client message
+	// client messages are sent by clients, like txgen, wallet
+	go node.receiveGroupMessage(node.clientReceiver)
+
+	// start the goroutine to receive group message
+	go node.receiveGroupMessage(node.shardGroupReceiver)
```
Comment (astralblue) [inline node/node.go]: TODO @harmony-ek – plumb multiple queues/worker pools; avoid starvation due to client flooding

File: msgq/msgq.go (hunk 2 above)
Comment (harmony-ek) [reply msgq/msgq.go]: Punted off, not immediately necessary (overruns are logged at least).

File: node/node.go (hunk 3 above)
Comment (harmony-ek) [reply node/node.go]: #1725 fixes this; merging from master to pick it up.

**Pull request merged**